### PR TITLE
Fix for REGISTRY-3233

### PR DESCRIPTION
**Pull request by Malintha**

### CODE REVIEW

File: apps/store/controllers/assets-router.jag [hunk 1]
```diff
@@ -74,14 +74,15 @@ var handleApiEndpoint = function (request, session, uriParams) {
 
 var handlePageEndpoint = function (request, session, uriParams) {
     var type = uriParams.type;
-    if(!validateType(type)){
+    var tenantId = getTenantId(request, session);
+    if (!validateType(type, tenantId)){
```
Comment (daneshk) [inline apps/store/controllers/assets-router.jag]: formatting issue

File: apps/store/controllers/assets-router.jag [hunk 2]
```diff
@@ -121,19 +121,29 @@ var server = require('store').server;
 var user = server.current(session);
 
 //Check if asset type is enabled
-var validateType = function (type){
-    //TODO: Tenant Id needs to be properly resolved or this will break when we do /t/
-    var tenantId = user ? user.tenantId : carbon.server.superTenant.tenantId;
+var validateType = function (type, tenantId){
```
Comment (daneshk) [inline apps/store/controllers/assets-router.jag]: formatting issue

File: apps/store/controllers/assets-router.jag (hunk 1 above)
Comment (Malintha) [inline apps/store/controllers/assets-router.jag]: Fixed in https://github.com/wso2/carbon-store/pull/843

File: apps/store/controllers/assets-router.jag (hunk 2 above)
Comment (Malintha) [inline apps/store/controllers/assets-router.jag]: Fixed in https://github.com/wso2/carbon-store/pull/843

File: apps/store/controllers/assets-router.jag [hunk 3]
```diff
@@ -121,19 +121,29 @@ var server = require('store').server;
...
     var availableTypes = rxt.app.getUIActivatedAssets(tenantId);
-
-            for (var index in availableTypes) {
-                if(availableTypes[index] === type){
-                    return true;
-                }
-            }
+    for (var index in availableTypes) {
+        if(availableTypes[index] === type){
```
Comment (Malintha) [inline apps/store/controllers/assets-router.jag]: Fixed in https://github.com/wso2/carbon-store/pull/843

**Pull request merged**